integrations/jenkins/core/client.py

```diff
@@ -0,0 +1,101 @@
+from urllib.parse import urlencode, quote
+
+import httpx
+from typing import Any, AsyncGenerator
+
+from loguru import logger
+
+
+class JenkinsClient:
+    def __init__(
+        self, jenkins_base_url: str, jenkins_user: str, jenkins_password: str
+    ) -> None:
+        self.jenkins_base_url = jenkins_base_url
+
+        auth = (jenkins_user, jenkins_password)
```

Comment:
can we use a token instead?

Reply:
either a token or password works. i will change the variable name to token though